proposals/0000-slurp.rst

```diff
@@ -0,0 +1,454 @@
+refuses the package upload. This is the only change necessary to
+``hackage-server``.
+
+It would be highly desirable that packages available elsewhere are also
+available to legacy cabal-install versions. To achieve this,
+``hackage-server`` is free to include ``.cabal`` files in the metadata
+bundle that correspond to package versions hosted elsewhere but
+registered via SLURP. Just like Stackage today imports package versions
+available from Hackage.
+
+6.4 No need for extra user accounts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+No user accounts are needed for making API calls to SLURP. SLURP is
+completely platform neutral. Platforms such as Hackage or Stackage can
+make SLURP API calls on behalf of users if they wish. They can choose to
+do so on the condition that users have an account on said platform, or
+not. Users can also make API calls directly.
+
+How can this work? The API does not permit changing any existing
+``pkgname->URL`` mapping in SLURP. Only adding new ones. Therefore, no
+existing package author can be adversely affected by the actions of any
+new package author.
+
+There is a danger of name squatting, i.e. users preemptively claiming
+names while not writing any package. This danger already exists for
+Hackage itself. The solution for dealing with that is exactly the same
+as for Hackage: define a policy for what constitutes a Haskell package,
+what constitutes name squatting, and appoint a small group of trustees
+to enforce the policy. Initially, the trustees can be the union of
+Hackage trustees and Stackage curators. No matter the chosen policy, the
+implementation of SLURP and the API remain the same.
+
+More than just name squatting, just like for Hackage, it’s conceivable
+that bad actors could run scripts to carpet bomb the package namespace.
+It’s easier to do so on SLURP than on Hackage (no user accounts). But in
+practice, this hasn’t been a major problem for other package managers
+with similar designs (e.g. Bower).
+
+7 Conclusion
+------------
+
+With SLURP, no service is *the* privileged place in the Haskell
+universe for hosting packages. Different actors in the community are
+free to build more package hosting services, without adversely affecting
+any of the existing ones or breaking backwards compatibility with
+existing build tools, all the while preserving two crucial properties:
+
+-  Package names remain independent of where they are hosted or who is
+   currently maintaining them.
+-  Package names uniquely and unambiguously map to a single endpoint
+   where the available versions can be listed and package tarballs
+   downloaded.
+
+This design is hardly new and has been validated on a larger scale than
+the Haskell community in the past. Bower for JavaScript (and also used
+by PureScript) has grown to about 60k packages, likewise with a registry
+of pointers to federated package repositories, and likewise doing so
+without requiring authentication. OPAM, the now dominant package manager
```

Comment:
The Bower approach did not scale well and was abandoned to the point that bower has now officially been deprecated. see https://github.com/bower/bower/issues/2298 and https://bower.io/blog/2017/how-to-migrate-away-from-bower/ among other locations. Bower is not a model to follow, endorse, or emulate.